src/main/java/com/aws/greengrass/deployment/IotJobsClientWrapper.java

```diff
@@ -172,85 +202,116 @@ public CompletableFuture<Integer> PublishDescribeJobExecution(DescribeJobExecuti
 
     @Override
     public CompletableFuture<Integer> SubscribeToDescribeJobExecutionAccepted(
-            DescribeJobExecutionSubscriptionRequest request, QualityOfService qos,
-            Consumer<DescribeJobExecutionResponse> handler, Consumer<Exception> exceptionHandler) {
+            DescribeJobExecutionSubscriptionRequest request,
+            QualityOfService qos,
+            Consumer<DescribeJobExecutionResponse> handler,
+            Consumer<Exception> exceptionHandler) {
 
         if (request.jobId == null || request.thingName == null) {
             CompletableFuture result = new CompletableFuture();
-            result.completeExceptionally(new MqttException(
-                    "DescribeJobExecutionSubscriptionRequest must have a non-null jobId and a non-null thingName"));
+            result.completeExceptionally(
+                    new MqttException(
+                            "DescribeJobExecutionSubscriptionRequest must have a non-null jobId and a non-null thingName"));
             return result;
         }
         String topic = String.format(JOB_DESCRIBE_ACCEPTED_TOPIC, request.thingName, request.jobId);
         Consumer<MqttMessage> messageHandler =
-                describeJobCbs.computeIfAbsent(new Pair<>(handler, exceptionHandler), (k) -> (message) -> {
-                    try {
-                        String payload = new String(message.getPayload(), StandardCharsets.UTF_8);
-                        DescribeJobExecutionResponse response =
-                                this.gson.fromJson(payload, DescribeJobExecutionResponse.class);
-                        handler.accept(response);
-                    } catch (Exception e) {
-                        if (exceptionHandler != null) {
-                            exceptionHandler.accept(e);
-                        }
-                    }
-                });
+                describeJobCbs.computeIfAbsent(
+                        new Pair<>(handler, exceptionHandler),
+                        (k) ->
+                                (message) -> {
+                                    try {
+                                        String payload =
+                                                new String(
+                                                        message.getPayload(),
+                                                        StandardCharsets.UTF_8);
+                                        DescribeJobExecutionResponse response =
+                                                this.gson.fromJson(
+                                                        payload,
+                                                        DescribeJobExecutionResponse.class);
+                                        handler.accept(response);
+                                    } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that your code handles a broad swath of exceptions in the catch block, potentially trapping dissimilar issues or problems that should not be dealt with at this point in the program.

[Learn more](https://cwe.mitre.org/data/definitions/396.html)